backtrace-library/src/main/java/backtraceio/library/common/ApplicationHelper.java

```diff
@@ -0,0 +1,74 @@
+package backtraceio.library.common;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class ApplicationHelper {
+    private static final transient String LOG_TAG = ApplicationHelper.class.getSimpleName();
+    /**
+     * Cached application name
+     */
+    private static String applicationName;
+
+    /**
+     * Cached application version
+     */
+    private static String applicationVersion;
+
+    /**
+     * Cached package name
+     */
+    private static String packageName;
+
+    /**
+     * Retrieves application name from context. The name will be cached over checks
+     * @param context application context
+     * @return application name
+     */
+    public static String getApplicationName(Context context) {
+        if(!BacktraceStringHelper.isNullOrEmpty(applicationName)) {
+            return applicationName;
+        }
+
+        applicationName = context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
+        return  applicationName;
+    }
+
+    /**
+     * Retrieves application version from the context. If the version name is not defined, the version code will be used instead.
+     * @param context application context
+     * @return current application version.
+     */
+    public static String getApplicationVersion(Context context) {
+        if(!BacktraceStringHelper.isNullOrEmpty(applicationVersion)) {
+            return applicationVersion;
+        }
+        try {
```

Comment:
if we want to avoid a try/catch, we can also use Gradle's BuildConfig.VERSION_CODE/BuildConfig.VERSION_NAME for applicationVersion.

Reply:
I didn't want to change this code proposed by @BartoszLitwiniuk. I also checked why it was implemented this way - it looks like we might not have access to the BuildConfig when we're working with context. Some info from the Internet:

Advantages of Using BuildConfig
- Simplicity: BuildConfig gives direct access to the version information without needing to call PackageManager or handle exceptions.
- Efficiency: Accessing BuildConfig is faster as it doesn’t require querying the system at runtime.

When to Use PackageManager Instead
The PackageManager approach is mainly useful if:

- You’re trying to retrieve the version information of another app installed on the device.
- You’re working with a context where BuildConfig might not be accessible (e.g., in some library modules).